src/snyk/common/services/learnService.ts

```diff
@@ -0,0 +1,151 @@
+import type { completeFileSuggestionType } from '../../snykCode/interfaces';
+import { OssIssueCommandArg } from '../../snykOss/views/ossVulnerabilityTreeProvider';
+import { ErrorHandler } from '../error/errorHandler';
+import { ILog } from '../logger/interfaces';
+
+const ecosystems = [
+  'all',
+  'java',
+  'javascript',
+  'python',
+  'golang',
+  'php',
+  'cpp',
+  'csharp',
+  'kubernetes',
+  'ruby',
+  'elixir',
+  'docker',
+  'npm',
+] as const;
+type LessonEcosystem = typeof ecosystems[number];
+
+const isValidEcosystem = (potentialEcosystem: string | null | undefined): potentialEcosystem is LessonEcosystem => {
+  return ecosystems.includes(potentialEcosystem as LessonEcosystem);
+};
+
+export type Lesson = {
+  title: string;
+  lessonId: string;
+  ecosystem: LessonEcosystem;
+  url: string;
+};
+
+type PackageMapper = {
+  [key in LessonEcosystem]?: string[];
+};
+
+export class LearnService {
+  private lessonsCache = new Map<
+    string,
+    {
+      data: Lesson[];
+      expiry: number;
+    }
+  >();
+
+  constructor(private readonly logger: ILog, private readonly shouldCacheRequests = true) {}
+
+  static getCodeIssueParams(issue: completeFileSuggestionType) {
+    const cwes = issue?.cwe || [];
+    const ecosystem = LearnService.convertCodeIdToEcosystem(issue.id);
+    return { cwes, ecosystem };
+  }
+
+  static getOSSIssueParams(issue: OssIssueCommandArg) {
+    const cwes = issue?.identifiers?.CWE || [];
+    const ecosystem = LearnService.convertOSSProjectTypeToEcosystem(issue.packageManager);
+    return { cwes, ecosystem };
+  }
+
+  static convertCodeIdToEcosystem(issueId: string): LessonEcosystem {
+    if (!issueId) return 'all';
+    const ecosystem = issueId.split(/\/|%2F/g)[0]; // %2F is the url encoding for /
+    if (isValidEcosystem(ecosystem)) {
+      return ecosystem;
+    } else {
+      return 'all';
+    }
+  }
+
+  static convertOSSProjectTypeToEcosystem(pkgManager: string): LessonEcosystem {
+    const snykLearnEcosystemToProjectTypeMapping: PackageMapper = {
+      javascript: ['npm', 'yarn', 'yarn-workspace'],
+      java: ['maven', 'gradle'],
+      python: ['pip', 'poetry', 'pipenv'],
+      csharp: ['nuget', 'paket'],
+      php: ['composer'],
+      ruby: ['rubygems'],
+      golang: ['golangdep', 'govendor', 'gomodules'],
+      elixir: ['hex'],
+    };
+    for (const [ecosystem, pkgManagers] of Object.entries(snykLearnEcosystemToProjectTypeMapping)) {
+      if (pkgManagers.includes(pkgManager) && isValidEcosystem(ecosystem)) {
+        return ecosystem;
+      }
+    }
+    return 'all';
+  }
+
+  async requestLessons(cwe: string) {
+    const cacheResult = this.lessonsCache.get(cwe);
+    if (this.shouldCacheRequests && cacheResult && cacheResult?.expiry > Date.now()) {
+      return cacheResult.data;
+    } else {
+      const res = await axios.get<Lesson[]>('/lessons', {
+        baseURL: 'https://api.snyk.io/v1/learn',
```

Comment:
It's better to declare URL as const and pull the host from `configuration.baseApiUrl` from configuration.ts